packages/web/server/queries/complex/route-token-out-given-in.ts

```diff
@@ -0,0 +1,127 @@
+import { Dec } from "@keplr-wallet/unit";
+import {
+  CONCENTRATED_LIQ_POOL_TYPE,
+  ConcentratedLiquidityPool,
+  ConcentratedLiquidityPoolRaw,
+  COSMWASM_POOL_TYPE,
+  CosmwasmPoolRaw,
+  FetchTickDataProvider,
+  OptimizedRoutes,
+  STABLE_POOL_TYPE,
+  StablePool,
+  StablePoolRaw,
+  Token,
+  TransmuterPool,
+  WEIGHTED_POOL_TYPE,
+  WeightedPool,
+  WeightedPoolRaw,
+} from "@osmosis-labs/pools";
+
+import { ChainList } from "~/config/generated/chain-list";
+
+import { queryNumPools } from "../osmosis";
+import { queryPaginatedPools } from "./pools";
+
+/**
+ * This function routes a given token to a specified output token denomination.
+ * It fetches the router, gets a quote for the route and retrieves candidate routes.
+ *
+ * @param token - The token to be routed.
+ * @param tokenOutDenom - The output token denomination.
+ * @returns Returns a promise that resolves with the quote and candidate routes. */
+export async function routeTokenOutGivenIn(
+  token: Token,
+  tokenOutDenom: string
```

Comment:
Same as above. I think we should just place all these params in objects. It makes it easier to identify the param and is more scalable. What do you think?